flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/common/utils/BigQueryPartition.java

```diff
@@ -0,0 +1,501 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.common.utils;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.flink.bigquery.services.PartitionIdWithInfo;
+import com.google.cloud.flink.bigquery.services.PartitionIdWithInfoAndStatus;
+import com.google.cloud.flink.bigquery.services.TablePartitionInfo;
+import org.apache.commons.lang3.StringUtils;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.TimeZone;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** Utility class to handle the BigQuery partition conversions to Flink types and structures. */
+@Internal
+public class BigQueryPartition {
```

Comment:
This should be named BigQueryPartitionUtils

Reply:
This file is hard to review since it also has some changes. Is it possible to move the file in a separate PR? Or move it after the review?

Reply:
Reverted file relocation. Just refactoring here.